src/analysis_feature.py

```diff
@@ -0,0 +1,214 @@
+"""
+Author: Zeliha Ural Merpez
+
+Date: Dec 04, 2020
+
+This script reads optimized model found from analysis.py.
+
+Usage: analysis_feature.py -i=<input> -o=<output> [-v]
+
+Options:
+-i <input>, --input <input>     Local raw data pkl filename and path
+-o <output>, --output <output>  Local output directory for created pngs
+[-v]                            Report verbose output of dataset retrieval process
+"""
+
+# example to run: python src/analysis_feature.py -i results/tables/optimized_model.pkl -o results
+
+import sys
+import os
+import numpy as np
+import pandas as pd
+import pickle
+from docopt import docopt
+from render_table import render_table
+from sklearn import datasets
+from sklearn.compose import (
+    ColumnTransformer,
+    TransformedTargetRegressor,
+    make_column_transformer,
+)
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.impute import SimpleImputer
+from sklearn.pipeline import Pipeline, make_pipeline
+from sklearn.preprocessing import OneHotEncoder, OrdinalEncoder, StandardScaler
+
+args = docopt(__doc__)
+
+def validate_inputs(input_file_path, output_dir_path):
+    """
+    Validates input argument paths.
+    Parameters:
+    -----------
+    input_file_path : str
+        input path to be verified
+
+    output_file_path : str
+        output path to be verified
+        
+    Returns:
+    -----------
+    None
+    """
+    if not os.path.isfile(input_file_path):
+        print(f"Cannot locate input file: {input_file_path}")
+        sys.exit()
+
+    if not os.path.exists(output_dir_path):
+        os.makedirs(output_dir_path)
+    if not os.path.exists(output_dir_path + "/figures"):
+        os.makedirs(output_dir_path + "/figures")
+    if not os.path.exists(output_dir_path + "/tables"):
+        os.makedirs(output_dir_path + "/tables")
+    assert os.path.exists(output_dir_path), f"Invalid output path: {output_dir_path}"
+    assert os.path.exists(output_dir_path + "/figures"), f"Invalid output path: {output_dir_path}/figures"
+    assert os.path.exists(output_dir_path + "/tables"), f"Invalid output path: {output_dir_path}/tables"
+
+
+def read_input_file(input_file_path):
+    """
+    Reads input file path and reads optimized model.
+    Parameters:
+    -----------
+    input_file_path : str
+        input path to be verified
+        
+    Returns:
+    -----------
+    float
+        The depth of bests model.
+    """
+    try:
+        with open(input_file_path, "rb") as pickle_file:
+            data_frame = pd.read_pickle(pickle_file)
+        if verbose: print('Input filename path is valid.')
+    except:
+        print(input_file_path + 'Input filename path is not valid. Please check!')
+        sys.exit()
+
+    best_depth = data_frame.loc[0, 'param_Random Forest Classifier__max_depth']
+    if verbose: print('Creating and returning optimized model depth.')
+    return best_depth
+
+def fit_best_model(data_frame, best_depth):
```

Comment:
Is this because the best model wasn't available? The best model should already be fit from analysis.py

Reply:
I have written this in case we change hyperparameters, I am using easy to read data frame pickle not pipeline pickle =)